src/main/java/roomescape/reservationTime/ReservationTimeService.java

```diff
@@ -1,49 +1,51 @@
 package roomescape.reservationTime;
 
+import java.time.LocalDate;
 import org.springframework.stereotype.Service;
 import roomescape.global.exception.RoomescapeBadRequestException;
 import roomescape.reservation.Reservation;
-import roomescape.reservation.ReservationDao;
 
 import java.util.List;
+import roomescape.reservation.ReservationRepository;
 
 @Service
 public class ReservationTimeService {
 
-    private ReservationTimeDao reservationTimeDao;
-    private ReservationDao reservationDao;
+    private final ReservationTimeRepository reservationTimeRepository;
+    private final ReservationRepository reservationRepository;
 
-    public ReservationTimeService(ReservationTimeDao reservationTimeDao, ReservationDao reservationDao) {
-        this.reservationTimeDao = reservationTimeDao;
-        this.reservationDao = reservationDao;
+    public ReservationTimeService(ReservationTimeRepository reservationTimeRepository,
+                                  ReservationRepository reservationRepository) {
+        this.reservationTimeRepository = reservationTimeRepository;
+        this.reservationRepository = reservationRepository;
     }
 
-    public List<AvailableTime> getAvailableTime(String date, Long themeId) {
-        List<Reservation> reservations = reservationDao.findAllReservationsByDateAndTheme(date, themeId);
-        List<ReservationTime> reservationTimes = reservationTimeDao.findAll();
+    public List<AvailableTime> getAvailableTime(LocalDate date, Long themeId) {
+        List<Reservation> reservations = reservationRepository.findByDateAndTheme_Id(date, themeId);
+        List<ReservationTime> reservationTimes = reservationTimeRepository.findAll();
 
         return reservationTimes.stream()
                 .map(time -> new AvailableTime(
                         time.getId(),
-                        time.getValue().toString(),
+                        time.getTimeValue().toString(),
                         reservations.stream()
                                 .anyMatch(reservation -> reservation.getTime().getId().equals(time.getId()))
```

Comment:
[퀴즈 ❓ ]
```suggestion
                                .anyMatch(reservation -> reservation.getTime().getId().equals(time.getId()))
```

이 로직을 실행할 때 추가 쿼리가 나갈까요? 나가지 않을까요?

ref) https://tecoble.techcourse.co.kr/post/2022-10-17-jpa-hibernate-proxy/

Reply:
아하 ! id를 초기화할 때는 프록시가 초기화되지 않아서 추가 쿼리가 나가지 않는군요 👍 

초기화 과정에서 id를 자바 빈 규약에 맞는 (getxxx) 로 호출하면 getIdentifier가 호출하게 되서 가지고 있던 id를 제공하는 것으로 이해했습니다. (내부 구조는 더 살펴봐야될 것 같아요 ㅜㅜ)

위에서 EAGER, LAZY 관해 학습할 때 fetchType에 따라 어떤 쿼리가 발생하는 지 테스트 코드 통해 확인했었어요. 
그때, xxx.getId() (`xxx = y.getxxxx()`)를 하면 추가 쿼리가 발생하지 않아 getTimeValue()로 변경해서 실험했었는 데 이런 이유가 있었군요 

감사합니다 !!

Reply:
현재 수준에서는 너무 깊이 이해하지 않아도 될 것 같습니다. 망고가 이해한대로 리뷰의 의도는
- 자바 빈 규약에 맞는 identifier 조회 (ex. getId)시 프록시가 초기화되지 않아 추가 쿼리가 날아가지 않는다
- identifier가 아닌 다른 필드값을 조회하려 시도할 때 추가쿼리가 날아간다.

정도만 캐치하셔도 100% 가져가신 거라 생각합니다 😄